integration-tests/smoke/ccip/ccip_token_transfer_test.go

```diff
@@ -0,0 +1,266 @@
+package smoke
+
+import (
+	"math/big"
+	"testing"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/stretchr/testify/require"
+
+	sel "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
+	testsetups "github.com/smartcontractkit/chainlink/integration-tests/testsetups/ccip"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/shared/generated/burn_mint_erc677"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+func TestTokenTransfer(t *testing.T) {
+	lggr := logger.TestLogger(t)
+	config := &changeset.TestConfigs{}
+	tenv, _, _ := testsetups.NewLocalDevEnvironmentWithDefaultPrice(t, lggr, config)
+	inMemoryEnv := false
+
+	// use this if you are testing locally in memory
+	// tenv := changeset.NewMemoryEnvironmentWithJobsAndContracts(t, lggr, 2, 4, config)
+	// inMemoryEnv := true
+
+	e := tenv.Env
+	state, err := changeset.LoadOnchainState(e)
+	require.NoError(t, err)
+
+	// Chain and account setup
+	allChainSelectors := maps.Keys(e.Chains)
+	sourceChain, destChain := allChainSelectors[0], allChainSelectors[1]
+	ownerSourceChain := e.Chains[sourceChain].DeployerKey
+	ownerDestChain := e.Chains[destChain].DeployerKey
+
+	// Deploy and fund self-serve actors
+	selfServeSrcTokenPoolDeployer := createAndFundSelfServeActor(t, ownerSourceChain, e.Chains[sourceChain], big.NewInt(1e18), inMemoryEnv)
+	selfServeDestTokenPoolDeployer := createAndFundSelfServeActor(t, ownerDestChain, e.Chains[destChain], big.NewInt(1e18), inMemoryEnv)
+
+	// Deploy tokens and pool by CCIP Owner
+	srcToken, _, destToken, _, err := changeset.DeployTransferableToken(
+		lggr,
+		tenv.Env.Chains,
+		sourceChain,
+		destChain,
+		ownerSourceChain,
+		ownerDestChain,
+		state,
+		e.ExistingAddresses,
+		"OWNER_TOKEN",
```

Comment:
This name is not relevant, to be honest, it could be named randomly because we never use that again. We use "owner_token" only as a documentation of what it does ;)